e2e/container/attach_test.go

```diff
@@ -0,0 +1,17 @@
+package container
+
+import (
+	"testing"
+
+	"github.com/gotestyourself/gotestyourself/icmd"
+)
+
+func TestAttachExitCode(t *testing.T) {
+	cName := "test-attach-exit-code"
+	icmd.RunCommand("docker", "run", "-d", "--rm", "--name", cName,
+		alpineImage, "sh", "-c", "sleep 5 ; exit 21").Assert(t, icmd.Success)
+	cmd := icmd.Command("docker", "wait", cName)
+	res := icmd.StartCmd(cmd)
+	icmd.RunCommand("docker", "attach", cName).Assert(t, icmd.Expected{ExitCode: 21})
+	icmd.WaitOnCmd(8, res).Assert(t, icmd.Expected{ExitCode: 0, Out: "21"})
```

Comment:
This is only waiting 8ms, I'm not sure if that's related to the failure.

I'll see if I can get this working

Reply:
This test fails for me locally as well so I'll prepare a patch to get it working. 

I think the `docker wait` command is unnecessary, all we need is a `poll.WaitOn()` for the container to be started.

Reply:
Didn't realize it was in `ms` :sweat_smile: 

Thanks `poll.WaitOn()` seems to be what I was looking for. I'll try updating the code tonight.

Reply:
I pushed a commit, it seems to be working now. I didn't use the WaitOn for container start. I'm not sure if this test can flake, maybe `runBackgroundContainsWithExitCode()` could use a WaitOn if we see it flake.